components/homme/src/share/derivative_mod_base.F90

```diff
@@ -1241,6 +1242,54 @@ function vlaplace_sphere_wk_contra(v,deriv,elem,var_coef,nu_ratio) result(laplac
     enddo
   end function vlaplace_sphere_wk_contra
 
+!DIR$ ATTRIBUTES FORCEINLINE :: second_order_findiff
+  function second_order_findiff(u1, u2, u3, zeta1, zeta3) result(du_dzeta)
+        real(kind=real_kind), intent(in) :: u1, u2, u3, zeta1, zeta3
+        real(kind=real_kind) :: du_dzeta
+        du_dzeta = (u2-u1) * zeta3**2.0_real_kind + (u3-u2) * zeta1**2.0_real_kind
+        du_dzeta = du_dzeta / (zeta1**2.0_real_kind * zeta3 - zeta1 * zeta3**2.0_real_kind)
```

Comment:
I'm still puzzled by the formula. Again, assuming zeta1=|x2-x1| and zeta3=|x3-x2|, shouldn't the denom have a + sign? I'm guessing this is derived by doing du_L = (u2-u1)/zeta1, du_R=(u3-u2)/zeta3, then doing a weighted avg of du_L and du_R, but if so, the denom should have a +. Here, the denom vanishes for zeta3=zeta1, which seems wrong. Am I missing something?

Reply:
@jhollowed @OkayHughes

Reply:
@bartgol This is an instance of the same sign confusion as in the other comment, right? The math seems to work out on my end.